internal/docstore/drivertest/drivertest.go

```diff
@@ -175,6 +172,8 @@ func RunConformanceTests(t *testing.T, newHarness HarnessMaker, ct CodecTester,
 	t.Run("DeleteQuery", func(t *testing.T) { withTwoKeyCollection(t, newHarness, testDeleteQuery) })
 	t.Run("UpdateQuery", func(t *testing.T) { withTwoKeyCollection(t, newHarness, testUpdateQuery) })
 
+	t.Run("BeforeDo", func(t *testing.T) { testBeforeDo(t, newHarness) })
```

Comment:
I suspect you will end up wanting to combine this with other `As` tests, but you can worry about that later.

Reply:
Agreed.